src/main/java/hudson/plugins/analysis/core/HealthAwareRecorder.java

```diff
@@ -521,16 +592,43 @@ public String getDefaultEncoding() {
     }
 
     /**
+     * @see {@link #getDefaultEncoding()}
+     */
+    @DataBoundSetter
+    public void setDefaultEncoding(String defaultEncoding) {
+        this.defaultEncoding = defaultEncoding;
+    }
+
+    /**
+     * Returns whether the current build uses maven.
+     *
+     * @param build
+     *            the current build
+     * @return <code>true</code> if the current build uses maven,
+     *         <code>false</code> otherwise
+     */
+    protected boolean isMavenBuild(final Run<?, ?> build) {
+        if (build instanceof AbstractBuild) {
+            return isMavenBuild((AbstractBuild) build);
+        } else {
+            return false;
+        }
+    }
+
+    /**
      * Returns whether the current build uses maven.
      *
      * @param build
      *            the current build
      * @return <code>true</code> if the current build uses maven,
      *         <code>false</code> otherwise
+     * @deprecated use {@link #isMavenBuild(Run)} instead
      */
+    @Deprecated
     protected boolean isMavenBuild(final AbstractBuild<?, ?> build) {
-        if (build.getProject() instanceof Project) {
-            Project<?, ?> project = (Project<?, ?>)build.getProject();
+        AbstractBuild aBuild = (AbstractBuild) build;
+        if (aBuild.getProject() instanceof Project) {
+            Project<?, ?> project = (Project<?, ?>) aBuild.getProject();
```

Comment:
This method is widely used in subclasses, and starting now is useful to have a way to call ```isMavenBuild``` passing a ```Run``` object - we are moving the ```instanceof``` here, better than have it in all implementing plugins.

Moreover, it was suggested by @jtnord.